Pre-existing TypeScript type errors in tsc --noEmit

## Description

Running `tsc --noEmit` reveals several type errors that predate recent work. These don't block `npm run check` (which runs eslint + vitest) but indicate type drift, likely from an Anthropic SDK version bump that changed some param types.

## Errors

**`src/agents/game-engine.ts:860`** and **`src/app.tsx:292`**
```
error TS2345: Argument of type '{}' is not assignable to parameter of type 'string'.
```

**`src/app.tsx:320`**
```
error TS2304: Cannot find name 'Anthropic'.
```
Missing type import — `Anthropic` is used as a type but not imported in that scope.

**`src/agents/subagent.test.ts:163,194`** and **`src/agents/subagents/dev-mode.test.ts:56`**
```
error TS2352: Conversion of type 'TextBlockParam' to type 'Record<string, unknown>' may be a mistake
```
Test mocks casting `TextBlockParam` to `Record<string, unknown>` — the SDK's `TextBlockParam` type likely tightened its index signature.

## Suggested fix

- Add missing `import type Anthropic from "@anthropic-ai/sdk"` where needed
- Fix the `{}` → `string` argument mismatches
- Update test casts to go through `unknown` first, or adjust mock shapes to match the current SDK types

--- a/src/agents/game-engine.ts
+++ b/src/agents/game-engine.ts
@@ -856,7 +856,7 @@ export class GameEngine {
       const { frontMatter: fm, body: fmBody, changelog: fmChangelog } = parseFrontMatter(currentSheet);
       if (fm.sheet_status === "complete") {
         delete fm.sheet_status;
-        const title = fm._title ?? characterName;
+        const title = String(fm._title ?? characterName);
         await this.fileIO.writeFile(filePath, serializeEntity(title, fm, fmBody, fmChangelog));
         const slug = characterName.toLowerCase().replace(/\s+/g, "-").replace(/[^a-z0-9-]/g, "");
         this.sceneManager.notifyEntityTouched(filePath, slug);

--- a/src/agents/subagent.test.ts
+++ b/src/agents/subagent.test.ts
@@ -160,7 +160,7 @@ describe("cacheSystemPrompt", () => {
     expect(blocks).toHaveLength(1);
     expect(blocks[0].type).toBe("text");
     expect(blocks[0].text).toBe("You are a summarizer.");
-    expect((blocks[0] as Record<string, unknown>).cache_control).toEqual({
+    expect((blocks[0] as unknown as Record<string, unknown>).cache_control).toEqual({
       type: "ephemeral",
       ttl: "1h",
     });
@@ -191,7 +191,7 @@ describe("oneShot", () => {
     expect(Array.isArray(call.system)).toBe(true);
     const blocks = call.system as Anthropic.TextBlockParam[];
     expect(blocks[0].text).toBe("Test prompt.");
-    expect((blocks[0] as Record<string, unknown>).cache_control).toEqual({
+    expect((blocks[0] as unknown as Record<string, unknown>).cache_control).toEqual({
       type: "ephemeral",
       ttl: "1h",
     });

--- a/src/agents/subagents/dev-mode.test.ts
+++ b/src/agents/subagents/dev-mode.test.ts
@@ -53,7 +53,7 @@ describe("buildDevPrompt", () => {
   it("returns TextBlockParam[] with cache_control on first block", () => {
     const blocks = buildDevPrompt("Test");
     expect(Array.isArray(blocks)).toBe(true);
-    expect((blocks[0] as Record<string, unknown>).cache_control).toEqual({ type: "ephemeral", ttl: "1h" });
+    expect((blocks[0] as unknown as Record<string, unknown>).cache_control).toEqual({ type: "ephemeral", ttl: "1h" });
   });
 
   it("includes campaign name", () => {

--- a/src/app.tsx
+++ b/src/app.tsx
@@ -1,5 +1,6 @@
 import React, { useState, useEffect, useCallback, useRef } from "react";
 import { Text, Box, useInput } from "ink";
+import type Anthropic from "@anthropic-ai/sdk";
 import { createClient } from "./config/client.js";
 import { readFileSync, writeFileSync, mkdirSync } from "node:fs";
 import { readFile, writeFile, appendFile, mkdir, readdir, stat, unlink, rmdir } from "node:fs/promises";
@@ -260,7 +261,7 @@ async function buildInitialSheet(
       // Mark sheet as complete so the DM doesn't re-promote
       const { frontMatter, body, changelog } = parseFrontMatter(updatedSheet);
       frontMatter.sheet_status = "complete";
-      const title = frontMatter._title ?? result.characterName;
+      const title = String(frontMatter._title ?? result.characterName);
       const tagged = serializeEntity(title, frontMatter, body, changelog);
       await io.writeFile(charPath, tagged);
     }

